pages/utilities/hash-generator.tsx

```diff
@@ -0,0 +1,274 @@
+import React, { useCallback, useMemo, useState } from "react";
+import { Textarea } from "@/components/ds/TextareaComponent";
+import PageHeader from "@/components/PageHeader";
+import { Card } from "@/components/ds/CardComponent";
+import { Button } from "@/components/ds/ButtonComponent";
+import { Label } from "@/components/ds/LabelComponent";
+import Header from "@/components/Header";
+import { useCopyToClipboard } from "@/components/hooks/useCopyToClipboard";
+import CallToActionGrid from "@/components/CallToActionGrid";
+import Meta from "@/components/Meta";
+import { CMDK } from "@/components/CMDK";
+import { Combobox } from "@/components/ds/ComboboxComponent";
+import { Input } from "@/components/ds/InputComponent";
+import crypto, { BinaryToTextEncoding } from "crypto";
+import { generateHash } from "@/components/utils/hash-generator.utils";
+import GitHubContribution from "@/components/GitHubContribution";
+
+const MAX_ITERATIONS = 50_000;
+
+type Algorithm =
+  | "sha256"
+  | "sha512"
+  | "md5"
+  | "pbkdf2"
+  | "hmac-sha256"
+  | "hmac-sha512";
+
+interface Option {
+  value: string;
+  label: string;
+}
+
+const algorithmOptions: Option[] = [
+  { value: "sha256", label: "SHA-256" },
+  { value: "sha512", label: "SHA-512" },
+  { value: "md5", label: "MD5" },
+  { value: "pbkdf2", label: "PBKDF2 (with SHA-512)" },
+  { value: "hmac-sha256", label: "HMAC-SHA-256" },
+  { value: "hmac-sha512", label: "HMAC-SHA-512" },
+];
+
+const encodingOptions: Option[] = [
+  { value: "hex", label: "Hex" },
+  { value: "base64", label: "Base64" },
+  { value: "latin1", label: "Latin1" },
+];
+
+export default function HashGenerator() {
+  const [textInput, setTextInput] = useState("");
+  const [algorithm, setAlgorithm] = useState<Algorithm>("sha256");
+  const [saltInput, setSaltInput] = useState("");
+  const [secretKey, setSecretKey] = useState("");
+  const [iterations, setIterations] = useState(10000);
+  const [outputLength, setOutputLength] = useState(64);
+  const [encoding, setEncoding] = useState<BinaryToTextEncoding>("hex");
+  const [hashOutput, setHashOutput] = useState("");
+  const { buttonText, handleCopy } = useCopyToClipboard();
+
+  const handleTextChange = useCallback(
+    (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+      setTextInput(event.currentTarget.value);
+    },
+    []
+  );
+
+  const handleSaltChange = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      setSaltInput(event.currentTarget.value);
+    },
+    []
+  );
+
+  const handleSecretKeyChange = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      setSecretKey(event.currentTarget.value);
+    },
+    []
+  );
+
+  const handleAlgorithmChange = useCallback((value: string) => {
+    setAlgorithm(value as Algorithm);
+  }, []);
```

Comment:
Instead of casting with `as` keyword, you can utilize `value: Algorithm`.
Please do the same of other functions where `as` is used.

```
  const handleAlgorithmChange = useCallback((value: Algorithm) => {
    setAlgorithm(value);
  }, []);
```